fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBOrderingQueryTest.java

```diff
@@ -0,0 +1,196 @@
+                assertEquals(Set.of("ordered_str_value"), plan.getUsedIndexes());
+            }
+            final List<Long> records = plan.execute(recordStore)
+                    .map(r -> r.getPrimaryKey().getLong(0))
+                    .asList().join();
+            if (hasIndex) {
+                TestHelpers.assertDiscardedNone(context);
+            }
+            return records;
+        }
+    }
+
+    protected void sortOnly(@Nullable String orderFunction, boolean withIndex, long... expected) throws Exception {
+        final RecordMetaDataHook hook = withIndex ? indexHook(orderFunction) : noIndexHook();
+        loadRecords(hook);
+        final RecordQuery query = RecordQuery.newBuilder()
+                .setRecordType("MySimpleRecord")
+                .setSort(fullOrderingKey(orderFunction))
+                .build();
+        final List<Long> actual = queryRecords(query, hook, withIndex);
+        assertEquals(Arrays.stream(expected).boxed().collect(Collectors.toList()), actual);
+    }
+
+    @Test
+    void testDefault() throws Exception {
+        sortOnly(null, true, 1, 2, 3, 4, 5, 6);
+    }
+
+    @Test
+    void testAscNullsFirst() throws Exception {
+        sortOnly("order_asc_nulls_first", true, 1, 2, 3, 4, 5, 6);
+    }
+
+    @Test
+    void testAscNullsLast() throws Exception {
+        sortOnly("order_asc_nulls_last", true, 1, 3, 4, 5, 2, 6);
+    }
+
+    @Test
+    void testDescNullsFirst() throws Exception {
+        sortOnly("order_desc_nulls_first", true, 1, 2, 5, 4, 3, 6);
+    }
+
+    @Test
+    void testDescNullsLast() throws Exception {
+        sortOnly("order_desc_nulls_last", true, 1, 5, 4, 3, 2, 6);
+    }
+
+    @Test
+    void testDescWithoutIndex() throws Exception {
+        sortOnly("order_desc_nulls_last", false, 1, 5, 4, 3, 2, 6);
+    }
+
+    protected void sortAndFilter(@Nullable String orderFunction, boolean withIndex, long... expected) throws Exception {
+        final RecordMetaDataHook hook = withIndex ? indexHook(orderFunction) : noIndexHook();
+        loadRecords(hook);
+        final RecordQuery query = RecordQuery.newBuilder()
+                .setRecordType("MySimpleRecord")
+                .setSort(fullOrderingKey(orderFunction))
+                .setFilter(Query.and(
+                        Query.field("num_value_2").equalsValue(101),
+                        Query.field("str_value_indexed").greaterThan("a")))
+                .build();
+        final List<Long> actual = queryRecords(query, hook, withIndex);
+        assertEquals(Arrays.stream(expected).boxed().collect(Collectors.toList()), actual);
+    }
+
+    @Test
+    void testFilteredDefault() throws Exception {
+        sortAndFilter(null, true, 4, 5);
+    }
+
+    @Test
+    void testFilteredAsc() throws Exception {
+        sortAndFilter("order_asc_nulls_first", true, 4, 5);
+    }
+
+    @Test
+    void testFilteredDesc() throws Exception {
+        sortAndFilter("order_desc_nulls_last", true, 5, 4);
```

Comment:
It might be good to check `order_desc_nulls_first` with this predicate to make sure it doesn't return the `null` `str_value_indexed`. Likewise with `testFilteredAsc`, it would be good to check it with `order_asc_nulls_last`

Reply:
I just went ahead and made it test all cases.